File: app/src/main/java/com/harry/harrypotter/ui/adapter/PersonAdapter.java
package com.harry.harrypotter.ui.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.harry.harrypotter.R;
import com.harry.harrypotter.model.Person;
import com.squareup.picasso.Picasso;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;


public class PersonAdapter  extends RecyclerView.Adapter<PersonAdapter.ViewHolder> {

    private LayoutInflater layoutInflater;
    private Context context;
    private PersonAdapter.ItemClickListener mClickListener;
    private List<Person> list;

    public PersonAdapter(Context context, List<Person> list) {
        this.context = context;
        this.layoutInflater = LayoutInflater.from(context);
        this.list = list;

    }

    @NonNull
    @Override
    public PersonAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = layoutInflater.inflate(R.layout.item_person, parent, false);
        return new PersonAdapter.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull PersonAdapter.ViewHolder holder, int position) {

        holder.name.setText(list.get(position).name);
        Picasso.get().load(list.get(position).image).into(holder.image);
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        ImageView image;
        TextView name;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            image = itemView.findViewById(R.id.image);
            name = itemView.findViewById(R.id.name);
            itemView.setOnClickListener(this);

        }

        @Override
        public void onClick(View view) {
            if (mClickListener != null) mClickListener.onItemClick(view, getAdapterPosition());
        }


    }

    public void setOnItemClickListener(PersonAdapter.ItemClickListener itemClickListener) {
        this.mClickListener = itemClickListener;
    }

    public interface ItemClickListener {
        void onItemClick(View view, int position);
    }




}

File: app/src/main/java/com/harry/harrypotter/ui/main/adapter/PersonAdapter.kt
package com.harry.harrypotter.ui.main.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.harry.harrypotter.databinding.ItemPersonBinding
import com.harry.harrypotter.model.Person
import com.squareup.picasso.Picasso

class PersonAdapter :
    RecyclerView.Adapter<PersonAdapter.ViewHolder>() {

    private var list: List<Person> = ArrayList()
    private lateinit var binding: ItemPersonBinding

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        binding = ItemPersonBinding.inflate(LayoutInflater.from(parent.context),
            parent, false)
        return ViewHolder(binding.root)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        binding.name.text = list[position].name
        Picasso.get().load(list[position].image).into(binding.image)
    }

    override fun getItemCount() = list.size

    fun setData(listOfPerson: List<Person>) {
        list = listOfPerson
        notifyDataSetChanged()
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)

}